src/scanspec/core.py

```diff
@@ -375,13 +377,16 @@ def __init__(
             f"Mismatching axes "
             f"{list(self.midpoints)} != {list(self.lower)} != {list(self.upper)}"
         )
-        # Check all lengths are the same
-        lengths = {
-            len(arr)
-            for d in (self.midpoints, self.lower, self.upper)
-            for arr in d.values()
-        }
-        lengths.add(len(self.gap))
+        if len(midpoints) > 0:
+            # Check all lengths are the same
+            lengths = {
+                len(arr)
+                for d in (self.midpoints, self.lower, self.upper)
+                for arr in d.values()
+            }
+            lengths.add(len(self.gap))
+        else:
+            lengths = {len(self.duration)} if self.duration is not None else {0}
```

Comment:
If duration and midpoints both exist we need to check that

```suggestion
        # Check all lengths are the same
        lengths = {
            len(arr)
            for d in (self.midpoints, self.lower, self.upper)
            for arr in d.values()
        }
        lengths.add(len(self.gap))
        if self.duration is not None:
            lengths.add(len(self.duration))
```

Reply:
Done!